packages/blockchain-wallet-v4-frontend/src/components/Flyout/Banners/UpgradeToGold.tsx

```diff
@@ -88,6 +89,23 @@ const UpgradeToGoldBanner = ({ limits, verifyIdentity }: Props) => {
     hideBanner((prevValue) => !prevValue)
   }
 
+  // find effective limit
+  let effectiveLimit = {} as LimitWithEffective
+  if (current?.daily?.effective) {
```

Comment:
This is a good use-case for a switch case.

Reply:
and/or a function that can go in a model file somewhere.

Reply:
yeah this is basically dirty working solution, will improve it